it/controllers/benefits/MileageBenefitAmountControllerISpec.scala

```diff
@@ -0,0 +1,513 @@
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CompanyCarFuelBenefitsController.show(taxYearEOY, employmentId).url)
+            }
+
+          }
+          "there is no data in session for that user" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              noUserDataStub(nino, taxYearEOY)
+              userDataStub(IncomeTaxUserData(None), nino, taxYearEOY)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlGet(url(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)), follow = false)
+            }
+
+            s"has an SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+            }
+
+          }
+
+          "it is not EOY" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              noUserDataStub(nino, taxYear)
+              insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true)), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlGet(url(taxYear), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), follow = false)
+            }
+
+            s"has an SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
+            }
+          }
+        }
+      }
+    }
+  }
+
+  ".submit" should {
+
+    import Selectors._
+
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "redirect" when {
+
+          "there is no cya data in session for that user" which {
+
+            lazy val form: Map[String, String] = Map(AmountForm.amount -> "200.00")
+
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true)), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlPost(url(taxYearEOY), body = form, user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            s"has a SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+              lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+              cyamodel.employment.employmentBenefits.flatMap(_.carVanFuelModel) shouldBe None
+            }
+          }
+          "the mileage question is no" which {
```

Comment:
do you need tests for all the different scenarios? or are you okay cause its the same method that the show uses?